# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/angle/checkout/src/libANGLE/MemoryObject.h
# Commit: 3a91885d47e0
# Full Hash: 3a91885d47e0ef7fba9749562d52300eea67c651
# Author: Miko Mynttinen <mikokm@gmail.com>
# Date: 2019-11-11 21:52:52
# Regressor Bug: 1578576
# File Overlap Count: 4
# Description:
#   Bug 1578576 - Part 2: Update to ANGLE 3865 r=jgilbert
#   
#   Depends on D44561
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D44579
# ==============================================================================

diff -r 77cee77c0307 -r 3a91885d47e0 gfx/angle/checkout/src/libANGLE/MemoryObject.h
--- /dev/null	Thu Jan 01 00:00:00 1970 +0000
+++ b/gfx/angle/checkout/src/libANGLE/MemoryObject.h	Fri Nov 08 17:06:00 2019 +0000
@@ -0,0 +1,45 @@
+//
+// Copyright 2019 The ANGLE Project Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+//
+// MemoryObject.h: Defines the gl::MemoryObject class [EXT_external_objects]
+
+#ifndef LIBANGLE_MEMORYOBJECT_H_
+#define LIBANGLE_MEMORYOBJECT_H_
+
+#include "angle_gl.h"
+#include "common/PackedEnums.h"
+#include "common/angleutils.h"
+#include "libANGLE/Error.h"
+#include "libANGLE/RefCountObject.h"
+
+namespace rx
+{
+class GLImplFactory;
+class MemoryObjectImpl;
+}  // namespace rx
+
+namespace gl
+{
+class Context;
+
+class MemoryObject final : public RefCountObject
+{
+  public:
+    MemoryObject(rx::GLImplFactory *factory, GLuint id);
+    ~MemoryObject() override;
+
+    void onDestroy(const Context *context) override;
+
+    rx::MemoryObjectImpl *getImplementation() const { return mImplementation.get(); }
+
+    angle::Result importFd(Context *context, GLuint64 size, HandleType handleType, GLint fd);
+
+  private:
+    std::unique_ptr<rx::MemoryObjectImpl> mImplementation;
+};
+
+}  // namespace gl
+
+#endif  // LIBANGLE_MEMORYOBJECT_H_